### PR TITLE
Jh.lj Force Unadopt Route for Admins

### PR DESCRIPTION
**Pull request by Tvcz**
@jhylisa666 and I worked on a new route that allows admins to force unadopt the sites of standard users and allows super admins to force unadopt the sites of standard users, admins, and other super admins.

In the course of modifying the ProtectedSiteProcessor, we also added a new utility method `isAdmin(PrivilegeLevel)` which returns a boolean indicating whether a user with the given privilege level is an admin or super admin.

Route is accessible via:
`POST api/v1/protected/sites/:site_id/force_unadopt`

### CODE REVIEW

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java [hunk 1]
```diff
@@ -119,6 +131,34 @@ public void unadoptSite(JWTData userData, int siteId) {
         .execute();
   }
 
+  @Override
+  public void forceUnadoptSite(JWTData userData, int siteId) {
+    isAdminCheck(userData.getPrivilegeLevel());
+    checkSiteExists(siteId);
+    if (!isAlreadyAdopted(siteId)) {
+      throw new WrongAdoptionStatusException(false); // wrong error?
```
Comment (SofieDunt) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:139]: 👍 This is a good exception to use, it'll return with message "Site is not marked as adopted", which makes sense

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java [hunk 2]
```diff
@@ -119,6 +131,34 @@ public void unadoptSite(JWTData userData, int siteId) {
...
+            .fetchInto(AdoptedSitesRecord.class)
+            .get(0);
+
+    Integer adopterId = adoptedSite.getUserId();
+
+    UsersRecord adopter = db.selectFrom(USERS)
+            .where(USERS.ID.eq(adopterId))
+            .fetchOne();
+
+    if(isAdmin(adopter.getPrivilegeLevel()) && !(userData.getPrivilegeLevel() == (PrivilegeLevel.SUPER_ADMIN))) {
```
Comment (SofieDunt) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:153]: ⛏️ Super small – this is using `==` for enum equality, but above it uses `.equals`. Both are correct, but we should keep it consistent

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java [hunk 3]
```diff
@@ -119,6 +131,34 @@ public void unadoptSite(JWTData userData, int siteId) {
         .execute();
   }
 
+  @Override
+  public void forceUnadoptSite(JWTData userData, int siteId) {
+    isAdminCheck(userData.getPrivilegeLevel());
+    checkSiteExists(siteId);
```
Comment (SofieDunt) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:137]: 👍

**Review by SofieDunt — APPROVED** [review]
💯 Looks good to me! Great work on this!

**Review by chromium-52 — APPROVED** [review]
Looks great! Could you just be able to add this route to the API spec page at https://github.com/Code-4-Community/c4c-dev-docs/blob/master/docs/sftt/sftt-api-spec.md?

**Pull request merged**